includes/specials/SpecialDeletedContributions.php

```diff
@@ -133,11 +142,17 @@ function getNamespaceCond() {
 	 */
 	function formatRow( $row ) {
 		wfProfileIn( __METHOD__ );
+
+		if ( is_numeric( $row->ar_user ) && $row->ar_user > 0 ) {
+			$userName = User::newFromId( $row->ar_user )->getName();
+		} else {
+			$userName = $row->ar_user_text;
+		}
 		$rev = new Revision( array(
 				'id'         => $row->ar_rev_id,
 				'comment'    => $row->ar_comment,
 				'user'       => $row->ar_user,
-				'user_text'  => $row->ar_user_text,
+				'user_text'  => $userName,
 				'timestamp'  => $row->ar_timestamp,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "timestamp" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)